File: aspn-c/src/aspn23/TypeIntegrity.c

/*
 * This code is generated via firehose.
 * DO NOT hand edit code.  Make any changes required using the firehose repo instead
 */

#include "TypeIntegrity.h"

Aspn23TypeIntegrity* ASPN_NULLABLE aspn23_type_integrity_new(uint8_t integrity_method,
                                                             double integrity_value) {
	Aspn23TypeIntegrity* self = (struct Aspn23TypeIntegrity*)malloc(sizeof(Aspn23TypeIntegrity));
	if (NULL == self) return NULL;
	self->integrity_method = integrity_method;
	self->integrity_value  = integrity_value;
	return self;
}

Aspn23TypeIntegrity* ASPN_NULLABLE aspn23_type_integrity_copy(Aspn23TypeIntegrity* input) {

	Aspn23TypeIntegrity* out =
	    aspn23_type_integrity_new(input->integrity_method, input->integrity_value);

	return out;
}

void aspn23_type_integrity_free(void* pointer) {
	Aspn23TypeIntegrity* self = (Aspn23TypeIntegrity*)pointer;
	if (NULL == self) return;
	aspn23_type_integrity_free_members(self);
	free(self);
}

void aspn23_type_integrity_free_members(Aspn23TypeIntegrity* self) {
	if (NULL == self) return;
}

File: aspn-py/src/aspn23/type_integrity.py
"""
This code is generated via firehose.
DO NOT hand edit code.  Make any changes required using the firehose repo instead.
"""


from dataclasses import dataclass
from typing import Optional

from numpy.typing import NDArray


@dataclass
class TypeIntegrity:
    """
    Integrity provides a numerical measure of the trustworthiness of the associated measurement. The
    integrity_method enum specifies the integrity method and the appropriate interpretation of the
    integrity_value.

    ### Attributes

    integrity_method - int:
            Enumerated field which describes the meaning of the integrity value. Integrity
            methods may use multiple enum entries if there is a need to pass multiple integrity
            values. For example, an integrity method with 3 parameters may use 3 enum entries (e.g.,
            MY_METHOD_VALUE1, MY_METHOD_VALUE2, MY_METHOD_VALUE3). When integrity is reported in a
            message, a vector of integrity types is passed. In the MY_METHOD example, num_integrity
            = 3 just for the MY_METHOD integrity. If additional integrity methods are also reported,
            then num_integrity would be greater than 3 to include those additional methods.

    integrity_value - Optional[Optional[float]]:
            Measurement integrity value to be interpreted based on the definition in the
            integrity_type enum.
    """

    integrity_method: int
    integrity_value: Optional[float]
